cpp/src/clients/python/radpy.cpp

```diff
@@ -1363,6 +1364,86 @@ static PyObject* radia_ObjDrwOpenGL(PyObject *self, PyObject *args)
 	return oInd;
 }
 
+/************************************************************************//**
+ * Magnetic Field Sources: Returns the OpenGL 3D geometry data
+ ***************************************************************************/
+static PyObject* radia_ObjGeometry_helper(const radGeometryFlattened &rgf) {
+    PyObject *lengths = PyList_New(rgf.Count);
+    PyObject *colors = PyList_New(rgf.Count * 3);
+    PyObject *vertices = PyList_New(rgf.VerticesCount * 3);
+    if (!vertices || !lengths || !colors) {
```

Comment:
I have a question: why do you sent these data to Py as Lists, and not as Py (native) arrays? Arrays have buffer interface and flat memory alignment, and can be manipulated in C/C++ via pointers, and are faster in Py too. We use arrays extensively in SRW for wavefront data (this would simply never work based on lists). I suggest changing this, if possible; though for the moment I can merge it ~as is.

Reply:
I remember looking into this, and I didn't see an example so I just used lists. Arrays are great. We can change our code if necessary.